collector/textfile.go

```diff
@@ -208,20 +210,39 @@ func (c *textFileCollector) Update(ch chan<- prometheus.Metric) error {
 		}
 
 		for _, f := range files {
+			metricsFilePath := filepath.Join(path, f.Name())
 			if !strings.HasSuffix(f.Name(), ".prom") {
 				continue
 			}
 
-			mtime, err := c.processFile(path, f.Name(), ch)
+			mtime, families, err := c.processFile(path, f.Name(), ch)
+			if families != nil {
```

Comment:
I just unblocked the missing lint CI check and it detected this error here:

```
 Error: S1031: unnecessary nil check around range (gosimple)
```

After fixing that, things should hopefully be good.

Reply:
Pretty cool these static checks! I got rid of the useless nil check, let's see.